### ISSUE TITLE
Remove attributes from terms

### ISSUE THREAD

**Issue by stasm** [projectfluent/fluent#98]
A possible solution for #97.

By removing the attributes from terms, we would completely remove the need for the `-term.attr` syntax. Thus, only two syntaxes would remain: `message.attr` and `-term[variant]`.

Metainformation about terms would need to be moved into variants:

```properties
-brand-name =
    {
       *[nominative] Firefox
        [accusative] Firefoxa
        [gender] masculine
    }
```

And would be accessed in the same way as other variants:

```properties
has-updated =
    { -brand-name[gender] ->
       *[masculine] { -brand-name[nominative] } został zaktualizowany.
        [feminine] { -brand-name[nominative] } została zaktualizowana.
        [neutral] { -brand-name[nominative] } zostało zaktualizowane.
    }
```

**Comment by zbraniecki** [comment]
That looks quite bad to me. I think of variants as a radiolist that selects one of the values. `gender` of `-brand-name` is not a radiolistvalue next to `accusative`. It's a completely different selection list.

I see this proposal a tradeoff attempting to squeeze a square box in a round hole to simplify taxonomy.

This proposal also changes the nature of fallbacking between variants/attributes. In the current design they fallback differently, and for a reason. If an attempt to retrieve a particular variant failed, it was a good `best-attempt` fallback to take the default variant, while if an attempt to format an attribute failed, getting "another attribute" was not.

To explain the concern, I'll use an extreme example:

```
-brand-name =
    {
        *[nominative] Firefox
         [accusative] Firefoksa
         [number] singular
         [gender] masculine
         [animacy] inanimate
    }
```

What is the default variant between `gender`, `nominative`, and `animacy`?

**Comment by stasm** [comment]
> What is the default variant between gender, nominative, and animacy?

I don't follow; can you rephrase please?

**Comment by stasm** [comment]
Removal of attributes would introduce a slight inconvenience for languages which don't require changes to the value but benefit from metainformation in attributes. Take German for example:

```properties
# With attributes
-brand-name = Firefox
    .Genus = Maskulinum
```

```properties
# Without attributes
-brand-name =
    {
       *[what should this be called?] Firefox
        [Genus] Maskulinum
    }
```

**Comment by Pike** [comment]
I'll have more comments in #97, but I've also got a couple specific comments that probably belong here and not there, so I'll put that here and reference it there.

The linter-check to find misuse of Variants in selectors or term-references would have to look like this:

> Find any variant of _Term_ that is used in at least one selector and at least one term-reference, and mark .... all references as errors?

I find that hard to compute.

Another linter concern: I think that it should be valid (no error, maybe warning) for a term-reference to ask for a non-existing variant of a _Term_. In particular if we support message references as external args, requiring the same Variant exposure of all Terms might not be right. OTH, a gender selector should really not fall through to the default variant of the term, I think.

In contrast, the two checks with the current variant-attribute split look like this:

> A variant can't be a selector in a well-formed file. An attribute can't be term-referenced in a well-formed file.

That allows also an easy way to do autocomplete here.

Second linter example, sadly still tricky:

> A reference to a non-existing attribute is at least a warning.

Most often, that's an error, but we'd recover by taking the default _Option_ of the select expression, and be OK.

> A reference to a non-existing variant might be a warning.

No good gut instinct on this one, we'd need more practical experience.

**Comment by stasm** [comment]
It looks like we all listed good reasons for not removing attributes from terms. I mostly filed this issue because I wanted to track what @Pike and I talked about in Berlin. Let's close it.
